density_scaler/src/server.rs

```diff
@@ -64,7 +64,14 @@ impl Server {
         }
         global_map.reduce_global();
         let new_map = compute_hex_density_map(&global_map);
+
+        tracing::info!("populated hex density map with {} keys", new_map.len());
+        let sample: Vec<(String, Decimal)> = new_map.clone().into_iter().take(3).collect::<Vec<(String, Decimal)>>();
```

Comment:
this may make sense, but is a reason to not do something along the lines of

```rust
new_map.clone().iter().take(3).cloned().collect::<Vec<(String, Decimal)>>();
```

?